### DISCUSSION TITLE
Automatic deployment via TheOpenEM on Windows machines

### DISCUSSION THREAD

**Discussion by Eruthon** [rustdesk/rustdesk#2865] [Show and tell]
I'd like to show a way to automatically deploy a RustDesk client via TheOpenEM management system:

1. create command module
- command: "rustdesk-\<version>-host=\<IP-or-DNS-of-server>,key=\<key>.exe"
- arguments: --reinstall
- external files:
  - File Name: rustdesk-\<version>-host=\<IP-or-DNS-of-server>,key=\<key>.exe
  - URL: https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.0-x86_64-pc-windows-msvc.exe

2. download your customized RustDesk2.toml via FileCopy module
- Destination: C:\\\<your-special-folder>\RustDesk
- Overwrite Existing Files
- Upload Files:
  - upload your customized RustDesk2.toml

3. create script module for importing custom config in RustDesk2.toml
- Working directory: C:\\\<your-special-folder>\RustDesk
- Script Contents:
`Restart-Service -Name RustDesk`
`Copy-Item "C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml" -Destination "C:\<your-special-folder>\Rustdesk"
& 'C:\Program Files\RustDesk\rustdesk.exe' --import-config Rustdesk.toml
& 'C:\Program Files\RustDesk\rustdesk.exe' --password <your-permanent-password>`
`Restart-Service -Name RustDesk`

For every one of your PCs, run this modules in their respective order or create a policy.
The script doesn't have any way to find out the remote ID, but you can use IP address, until there is a way to obtain the ID.

**Comment by rustdesk** [comment]
Good job. Thanks @Eruthon . We will add more cmd line support in the future, but as for what arguments should be supported, we need some investigations.

**Reply by Eruthon** [reply]
Thanks. But regarding the ID, if the Address Book will be implemented and the self-host server would collect the IDs from its clients, so I could, as an admin, log in my Rustdesk client and administrate any PC I need, it's not too important for this temporary solution.
Either way, great work so far! I love this piece of software!
